vault/identity_store_entities.go

```diff
@@ -756,6 +780,33 @@ func (i *IdentityStore) mergeEntity(ctx context.Context, txn *memdb.Txn, toEntit
 			return errors.New("entity id to merge from does not belong to this namespace"), nil
 		}
 
+		// If we're not resolving a conflict, we check to see if
+		// any aliases conflict between the toEntity and this fromEntity:
+		if !forceMergeAliases && len(conflictingAliasIDsToKeep) == 0 {
```

Comment:
If we're not forcing the merge, it seems like the alias clashes will never be checked or populated. Could you explain why we're doing this?

Reply:
Essentially, we have two 'modes':
- print error if there are clashes
- merge clashes (if conflictingAliasIDsToKeep is nonempty)

Essentially - if we're in 'merge clashes mode', we don't want to print errors about clashes that could be resolved, we want to print errors by any clashes that aren't resolved by the set of aliasIds presented. To rephrase: want to attempt the merge, instead of warn them that they need to select the aliases they want to keep to perform it.

This is a bit confusing, so I'm not sure if there's a way to make it clearer.